### PR TITLE
Pm 929 Time Validity of the requestId

### PR DESCRIPTION
**Pull request by DavideAmatoNexi**
#### List of Changes

`requestId` made valid only for a certain amount of time.
Added JwtToken for vpos, xpay and postepay

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have created or updated an API interface.
  - [ ] I have updated the OpenAPI spec
- [ ] I have created or updated a database table
  - [ ] I have updated both the Oracle and Postgres migrations
- [x] My PR is cohesive and self-contained (rule of thumb: less than 250 changed lines) 

### Link to story

https://pagopa.atlassian.net/browse/PM-929

### CODE REVIEW

**Comment by giovanniberti** [comment]
Please update the issue title to make it more descriptive

File: src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java [hunk 1]
```diff
@@ -92,6 +95,9 @@ public class PostePayPaymentTransactionsController {
     @Autowired
     private Environment environment;
 
+    @Autowired
+    private JwtTokenUtils jwtTokenUtils;
+
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java:100]: Prefer constructor injection to field injection

File: src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java [hunk 2]
```diff
@@ -403,8 +409,9 @@ private ResponseEntity<PostePayAuthResponse> createPostePayAuthResponse(String c
         postePayAuthResponse.setChannel(channel);
         postePayAuthResponse.setRequestId(requestId);
         postePayAuthResponse.setCorrelationId(correlationId);
+        String sessionToken = jwtTokenUtils.generateToken(requestId);
         if (StringUtils.isEmpty(errorMessage)) {
-            String urlRedirect = String.format(pgsResponseUrlRedirect, requestId, Scopes.POSTEPAY_SCOPE);
+            String urlRedirect = StringUtils.join(String.format(pgsResponseUrlRedirect, requestId, Scopes.POSTEPAY_SCOPE), "#token=", sessionToken);
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java:414]: If you expect this strings to be not null prefer using `+` to concatenate strtings

File: src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java [hunk 3]
```diff
@@ -72,6 +73,9 @@ public class XPayPaymentController {
     @Autowired
     private XPayUtils xPayUtils;
 
+    @Autowired
+    private JwtTokenUtils jwtTokenUtils;
+
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java:78]: Prefer constructor injection to field injection

File: src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java [hunk 4]
```diff
@@ -153,7 +157,8 @@ private ResponseEntity<XPayAuthResponse> createXpayAuthResponse(String errorMess
         }
 
         if (StringUtils.isEmpty(errorMessage)) {
-            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId);
+            String sessionToken = jwtTokenUtils.generateToken(requestId);
+            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId, "#token=", sessionToken);
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java:161]: If you expect this strings to be not null prefer using `+` to concatenate strtings

File: src/main/java/it/pagopa/pm/gateway/service/VposService.java [hunk 5]
```diff
@@ -186,7 +188,8 @@ private StepZeroResponse createStepZeroResponse(String errorMessage, String requ
         }
 
         if (StringUtils.isEmpty(errorMessage)) {
-            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId);
+            String sessionToken = jwtTokenUtils.generateToken(requestId);
+            String urlRedirect = StringUtils.join(responseUrlRedirect, requestId, "#token=", sessionToken);
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/pm/gateway/service/VposService.java:192]: Same as above

**Review by giovanniberti — CHANGES_REQUESTED** [review]

File: src/main/java/it/pagopa/pm/gateway/service/VposService.java (hunk 5 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/service/VposService.java]: Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/2ea78190d0495b5dfadfdafe1a469312ce6be9a0

File: src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java (hunk 4 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java]: Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/2ea78190d0495b5dfadfdafe1a469312ce6be9a0

File: src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java (hunk 2 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/controller/PostePayPaymentTransactionsController.java]: Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/2ea78190d0495b5dfadfdafe1a469312ce6be9a0

File: src/test/java/it/pagopa/pm/gateway/beans/ValidBeans.java [hunk 6]
```diff
@@ -251,7 +251,7 @@ public static PostePayAuthResponse postePayAuthResponse(String clientId, boolean
         if (isError) {
             postePayAuthResponse.setError(errorMessage);
         } else {
-            postePayAuthResponse.setUrlRedirect("${postepay.pgs.response.urlredirect}" + "#token=");
+            postePayAuthResponse.setUrlRedirect("${postepay.pgs.response.urlredirect}" + "#token=" + null);
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/pm/gateway/beans/ValidBeans.java:254]: Is this correct? The resulting string contains `#token=null` at the end

File: src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java (hunk 3 above)
Comment (DavideAmatoNexi) [reply src/main/java/it/pagopa/pm/gateway/controller/XPayPaymentController.java]: Done https://github.com/pagopa/pagopa-payment-transactions-gateway/pull/210/commits/346e0aed4f064be62656a816027f182d0bef5fdd

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**